data-access-services/lib/src/main/java/at/deloittedigital/core_banking/lib/entity/BaseEntity.java

```diff
@@ -0,0 +1,30 @@
+package at.deloittedigital.core_banking.lib.entity;
+
+import java.time.LocalDateTime;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+@Getter
+@Setter
+public abstract class BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @CreatedDate
+    private LocalDateTime createdAt;
```

Comment:
Do we need to define these fields in the db schema as well? Afaik currently we do not have these fields for tracking changes.

Reply:
Depends how we want to handle the schema generation:
- Manually, then yes, we should add createdAt and updatedAt to all tables as it greatly improves debugging imo. We also need to set `spring.jpa.hibernate.ddl-auto` to `validate`
- Let Hibernate handle it and set `spring.jpa.hibernate.ddl-auto` to `update/create/create-drop`
- Use Liquibase or Flyway for database migrations (similar to doing it manually)

Reply:
Both for local testing and for AWS we will have data already setup in MVP0 (for local the container db init script will do it, in AWS the db is fully managed in RDS using the same schema init script), so we will use `validate` I assume, and do not modify the schema directly from the spring boot services.

Can you pls align with @kanjula-archana about the necessary schema changes if we do want to support these audit fields? Thanks!

Reply:
Will do, thanks!